### PR TITLE
Introduce Support for PUSH0 

### PR DESCRIPTION
**Pull request by LuisPH3**
This PR adds PUSH0 and with it Shanghai support for CT, lfvm and evmzero. 

- [x]  unit tests
- [x]  ct evmzero
- [ ]  ct lfvm
- [ ]  full ct

### CODE REVIEW

File: cpp/vm/evmzero/interpreter_test.cc [hunk 1]
```diff
@@ -4348,11 +4349,35 @@ TEST(InterpreterTest, JUMPDEST_OutOfGas) {
...
+// PUSH0
+TEST(InterpreterTest, PUSH0) {
+  RunInterpreterTest({
+      .code = {op::PUSH0},
+      .state_after = RunState::kDone,
+      .gas_before = 10,
+      .gas_after = 8,
+      .stack_before = {},
+      .stack_after = {0x0},
+      .revision = EVMC_SHANGHAI,
```
Comment (LuisPH3) [inline cpp/vm/evmzero/interpreter_test.cc:4362]: @HerbertJordan This test checks push0 for Shangai, the rest of the test has baseline Istanbul.

**Review by facuMH — CHANGES_REQUESTED** [review]
looks good. just some minor details

File: go/ct/spc/specification.go [hunk 2]
```diff
@@ -691,10 +689,39 @@ func getAllRules() []Rule {
 		op:        JUMPDEST,
 		staticGas: 1,
 		pops:      0,
-		pushes:    0,
-		effect:    noEffect,
+		pushes:    1,
```
Comment (LuisPH3) [inline go/ct/spc/specification.go:692]:
```suggestion
		pushes:    0,
```

File: cpp/vm/evmzero/interpreter.cc [hunk 3]
```diff
@@ -1287,9 +1303,8 @@ struct PushImpl {
 };
 
 template <op::OpCode op_code>
-requires(OpCode::PUSH1 <= op_code && op_code <= OpCode::PUSH32)  //
-    struct Impl<op_code> : PushImpl<static_cast<uint64_t>(op_code - OpCode::PUSH1 + 1)> {
-};
+  requires(OpCode::PUSH1 <= op_code && op_code <= OpCode::PUSH32)  //
```
[the remark this answers is not in the document]
Comment (LuisPH3) [reply cpp/vm/evmzero/interpreter.cc]: not that easy, this formatting requires clang-format-14
anyways... we will bump up the formatting tool some day.

File: go/vm/lfvm/opcode.go [hunk 4]
```diff
@@ -23,6 +23,8 @@ type OpCode uint16
 const (
 	// Stack operations
 	POP OpCode = iota
+	PUSH0
+
```
[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/vm/lfvm/opcode.go:27]: I want to separate push0 from the rest, just like other ops down the file.

**Comment by LuisPH3** [comment]
this PR is has been replaced by:
#481, #485, and #486

**Pull request closed without merging**